hmpy/views/switch.py

```diff
@@ -0,0 +1,147 @@
+            :return:
+            """
+            paint = QPainter()
+
+            # Get view size and center
+            size = rect.size()
+
+            # Draw graphics to image (double buffer)
+            image = QImage(size, QImage.Format_ARGB32_Premultiplied)
+            image.fill(0)
+
+            paint.begin(image)
+            paint.translate(self.width() / 4, 0)
+
+            # Set the border gradient
+            border_gradient = QConicalGradient(rect.center(), -30)
+            border_color = self.palette().color(QPalette.Dark)
+            border_gradient.setColorAt(0.2, border_color)
+            border_gradient.setColorAt(0.5,
+                                       self.palette().color(QPalette.Light))
+            border_gradient.setColorAt(0.8, border_color)
+
+            paint.setBrush(border_gradient)
+            scale_x = self.width() / self._SCALE
+            scale_y = self.height() / self._SCALE
+            paint.drawRect(-20 * scale_x, 0,
+                           85 * scale_x, 85 * scale_y)
+
+            paint.setBrush(Qt.darkGray)
+            paint.drawRect(-5 * scale_x, 10 * scale_y,
+                           55 * scale_x, 70 * scale_y)
+
+            if self.isChecked():
+                paint.setBrush(Qt.green)
+                paint.drawRect(-5 * scale_x, 10 * scale_y,
+                               55 * scale_x, 35 * scale_y)
+                paint.setBrush(Qt.darkGreen)
+                paint.drawRect(-5 * scale_x, 8 * scale_y,
+                               55 * scale_x, 10 * scale_y)
+            else:
+                paint.setBrush(Qt.red)
+                paint.drawRect(-5 * scale_x, 45 * scale_y,
+                               55 * scale_x, 35 * scale_y)
+
+                paint.setBrush(Qt.darkRed)
+                paint.drawRect(-5 * scale_x, 70 * scale_y,
+                               55 * scale_x, 10 * scale_y)
+
+            paint.save()
+
+            paint.translate(self.width() / 2, 35 * scale_y)
+
+            paint.setBrush(Qt.darkGray)
+            paint.drawRect(-70 * scale_x, 50 * scale_y,
+                           85 * scale_x, 12 * scale_y)
+
+            paint.setPen(Qt.white)
+
+            paint.drawText(-55 * scale_x, 58 * scale_y, self._text)
+
+            paint.restore()
+            paint.end()
+
+            # Convert to pixmap and return
+            return QPixmap.fromImage(image)
+
+    def __init__(self, text, on=False):
+        """Initialize the switch view.
+
+        :param on: the state of the button"""
+        super().__init__()
+        self._button = self._SwitchButton(text, self)
+        self._button.setChecked(on)
+
+    @property
+    def on(self):
+        return self._button.isChecked()
+
+    @on.setter
+    def on(self, state):
+        """
+        Set the button state
+
+        :param state:
+        :return:
+        """
+        if state != self._button.isChecked():
+            self._button.toggle()
```

Comment:
Small note, but when you provide your own implementation to abstract something lower (self.toggle()), you should reuse that whenever possible.  That way if you ever needed to change what toggling the button does behind the scenes you only have to make the changes in one place

Reply:
Similarly, In the constructor you should set the on property instead of calling self._button.setChecked